TODO.md

```diff
@@ -9,5 +9,17 @@
 
 - Medium-Hard: Ability to modify existing entries
 
+  DONE - every editable element must have a 'edit' button (a little pencil)
+  DONE - Move Modal and ModalWrapper into the UI folder
+  DONE - Edit Modal existing prop names and methods so they are agnostic about the parent component
+
+  - Add new props to the Modal component so it can be repurposed:
+    DONE - modal title
+    DONE - button label
+    ???????? - initial value (must be "" when adding a new element, while it should show the current value of an element while editing it )
+    DONE - create a new EditElemModal component, and use the updated Modal component to show an 'Edit Modal'
+    DONE - clicking on the edit button on each element should show the "Edit modal", both on Menu and Active list
+  - TO BE CONTINUED ====>1
+

```

Comment:
Puoi rimuovere questo blocco e scrivere:

- add typescript
- add contexts
- refactor some code in custom hooks
- delete confirmation modal

Sono le cose che ti mostrero!